talk/tools.tex

```diff
@@ -81,6 +81,47 @@
   \end{minted}
 \end{frame}
 
+\subsection[format]{Code formatting}
+
+\begin{frame}[fragile]
+\frametitle{clang-format}
+\begin{block}{.clang-format}
+	\begin{itemize}
+		\item file describing your formatting preferences
+		\item should be checked-in at the repository root (project wide)
+		\item \mintinline{bash}{clang-format -style=LLVM -dump-config >} \\
+		  \mintinline{bash}{.clang-format}
+		\item adapt style options with help from: \url{https://clang.llvm.org/docs/ClangFormatStyleOptions.html}
+	\end{itemize}
+\end{block}
+\begin{block}{Run clang-format}
+	\begin{itemize}
+		\item \mintinline{bash}{clang-format --style=LLVM -i <file.cpp>}
+		\item \mintinline{bash}{clang-format -i <file.cpp>} (looks for .clang-format file)
+		\item \mintinline{bash}{git clang-format} (formats local changes)
+		\item \mintinline{bash}{git clang-format <ref>} (formats changes since git \textless{}ref\textgreater{})
+		\item Some editors/IDEs find a .clang-format file and adapt
+	\end{itemize}
+\end{block}
+\end{frame}
+
+\begin{frame}[fragile]
+\frametitle{clang-format}
+\begin{alertblock}{Exercise Time}
+	\begin{itemize}
+		\item go to any example
+		\item format code with: \mintinline{bash}{clang-format --style=GNU -i <file.cpp>}
+		\item inspect changes, try \mintinline{bash}{git diff}
+		\item revert changes using \mintinline{bash}{git checkout <file.cpp>}
```

Comment:
```suggestion
		\item revert changes using \mintinline{bash}{git checkout -- <file.cpp>}
```
?
In my git, I need the `--` to really check out a file. It might otherwise conflict with branch names.

Reply:
I just tried that and both work. I can do:
`git checkout tools.tex`
`git checkout -- tools.tex`
Both seem to work for me. But I indeed wondered then what would happen if I had a branch named `tools.tex`, so I tried it, and your solution keeps working. I will change the slides.

Reply:
Done.